channeldb/payment_control.go

```diff
@@ -730,11 +729,6 @@ func ensureInFlight(payment *MPPayment) error {
 	case paymentStatus == StatusInFlight:
 		return nil
 
-	// Our records show the payment as unknown, meaning it never
-	// should have left the switch.
-	case paymentStatus == StatusUnknown:
-		return ErrPaymentNotInitiated
-
 	// The payment succeeded previously.
 	case paymentStatus == StatusSucceeded:
```

Comment:
It looks like this list of cases is exhaustive now, so maybe the `default` case and `ErrUnknownPaymentStatus` can be removed?

Reply:
i think it's better to be explicit and return specific errors.